src/Process/ObjectRemovalHelper.js

```diff
@@ -68,9 +68,13 @@ export default {
      * @return {Array} an array of removed Object3D from obj (not including the recursive removals)
      */
     removeChildrenAndCleanupRecursively(layer, obj) {
-        let toRemove = obj.children.filter(c => (c.layer && c.layer.id) === layer.id);
+        let toRemove = obj.children.filter(c => c.layer && c.layer.id === layer.id);
```

Comment:
Shouldn't all this filtering logic on layers be removed since we don't set layers anymore?

Reply:
I just saw the modifications in TileDebug.js so I understand why you added this logic. Could you move it to a dedicated function like `getLinkedDebugObjects` for instance so it is clearer why we do all this logic here?

Reply:
> Shouldn't all this filtering logic on layers be removed since we don't set layers anymore?

yes, it seems there's cleaning to do
i'll try to remove some parts of this file

Reply:
Yes, but mind [this PR](https://github.com/iTowns/itowns/pull/1886) that modifies this file too. Maybe we should merge it and you should rebase before cleaning?

Reply:
> Yes, but mind [this PR](https://github.com/iTowns/itowns/pull/1886) that modifies this file too. Maybe we should merge it and you should rebase before cleaning?

yes, I'll rebase this PR

Reply:
> what part of the code would you like to move into this function?

My comment is misplaced sorry, I meant the part just below (from line 72 to 77) could be moved to a separate function.

Reply:
I thought about this `ObjectRemovalHelper.js` and I would like remove all nodes in their layer. So, it would be  possible to remove `node.layer.id === layer.id` in this file. (In same case, if `links` are grouped by `layer` it wouldn't have this test)

However, `LabelLayer` needs to be extend by `FeatureGeometryLayer`. In bonus we would simplify  label rendering!

All these changes need a new PR

Reply:
Yes good idea! Do you think you will be able to it soon and remember it or should we open an issue about it ?

Reply:
I opened #1991